x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_creation/helpers.ts

```diff
@@ -100,215 +100,114 @@ export interface RuleFields {
-  | 'threatQueryBar'
-  | 'threatMapping'
-  | 'eqlOptions'
-  | 'newTermsFields'
-  | 'historyWindowSize'
->;
-type EqlQueryRuleFields<T> = Omit<
-  T,
-  | 'anomalyThreshold'
-  | 'machineLearningJobId'
-  | 'threshold'
-  | 'threatIndex'
-  | 'threatQueryBar'
-  | 'threatMapping'
-  | 'newTermsFields'
-  | 'historyWindowSize'
->;
-type ThresholdRuleFields<T> = Omit<
-  T,
-  | 'anomalyThreshold'
-  | 'machineLearningJobId'
-  | 'threatIndex'
-  | 'threatQueryBar'
-  | 'threatMapping'
-  | 'eqlOptions'
-  | 'newTermsFields'
-  | 'historyWindowSize'
->;
-type MlRuleFields<T> = Omit<
-  T,
-  | 'queryBar'
-  | 'index'
-  | 'threshold'
-  | 'threatIndex'
-  | 'threatQueryBar'
-  | 'threatMapping'
-  | 'eqlOptions'
-  | 'newTermsFields'
-  | 'historyWindowSize'
->;
-type ThreatMatchRuleFields<T> = Omit<
-  T,
-  | 'anomalyThreshold'
-  | 'machineLearningJobId'
-  | 'threshold'
-  | 'eqlOptions'
-  | 'newTermsFields'
-  | 'historyWindowSize'
->;
-type NewTermsRuleFields<T> = Omit<
-  T,
-  | 'anomalyThreshold'
-  | 'machineLearningJobId'
-  | 'threshold'
-  | 'threatIndex'
-  | 'threatQueryBar'
-  | 'threatMapping'
-  | 'eqlOptions'
->;
-
-const isMlFields = <T>(
-  fields:
-    | QueryRuleFields<T>
-    | EqlQueryRuleFields<T>
-    | MlRuleFields<T>
-    | ThresholdRuleFields<T>
-    | ThreatMatchRuleFields<T>
-    | NewTermsRuleFields<T>
-): fields is MlRuleFields<T> => has('anomalyThreshold', fields);
-
-const isThresholdFields = <T>(
-  fields:
-    | QueryRuleFields<T>
-    | EqlQueryRuleFields<T>
-    | MlRuleFields<T>
-    | ThresholdRuleFields<T>
-    | ThreatMatchRuleFields<T>
-    | NewTermsRuleFields<T>
-): fields is ThresholdRuleFields<T> => has('threshold', fields);
-
-const isThreatMatchFields = <T>(
-  fields:
-    | QueryRuleFields<T>
-    | EqlQueryRuleFields<T>
-    | MlRuleFields<T>
-    | ThresholdRuleFields<T>
-    | ThreatMatchRuleFields<T>
-    | NewTermsRuleFields<T>
-): fields is ThreatMatchRuleFields<T> => has('threatIndex', fields);
-
-const isNewTermsFields = <T>(
-  fields:
-    | QueryRuleFields<T>
-    | EqlQueryRuleFields<T>
-    | MlRuleFields<T>
-    | ThresholdRuleFields<T>
-    | ThreatMatchRuleFields<T>
-    | NewTermsRuleFields<T>
-): fields is NewTermsRuleFields<T> => has('newTermsFields', fields);
-
-const isEqlFields = <T>(
-  fields:
-    | QueryRuleFields<T>
-    | EqlQueryRuleFields<T>
-    | MlRuleFields<T>
-    | ThresholdRuleFields<T>
-    | ThreatMatchRuleFields<T>
-    | NewTermsRuleFields<T>
-): fields is EqlQueryRuleFields<T> => has('eqlOptions', fields);
-
-export const filterRuleFieldsForType = <T extends Partial<RuleFields>>(
-  fields: T,
-  type: Type
-):
-  | QueryRuleFields<T>
-  | EqlQueryRuleFields<T>
-  | MlRuleFields<T>
-  | ThresholdRuleFields<T>
-  | ThreatMatchRuleFields<T>
-  | NewTermsRuleFields<T> => {
+const filterRuleFieldsWrappedWithType = <T extends Partial<RuleFields>>(fields: T, type: Type) => {
   switch (type) {
     case 'machine_learning':
-      const {
-        index,
```

Comment:
I did refactor how the property omission is done here, let me know your thoughts. I can file this as a separate PR if you feel like this would work better.